dg.go

```diff
@@ -223,25 +223,43 @@ func (d *directedGraph[NodeType]) PushStartingNodes() error {
 	defer d.lock.Unlock()
 
 	for nodeID, n := range d.nodes {
+		// Search for nodes that have either no outstanding dependencies, or only
+		// obviated dependencies.
 		if len(n.outstandingDependencies) == 0 {
 			d.readyForProcessing[nodeID] = n
+		} else {
+			// Has nodes left. Check to see if they are all obviated.
+			hasNonObviatedDependencies := false
```

Comment:
You know, it's a little odd that you pick a very long and descriptive name for this local variable but use just `n` for the actual node. (Which my brain keeps trying to identity as a simple integer, just like it would for `i`.)

This name doesn't need to be quite so big, like maybe `isReady`, which it is if it has only obviated dependencies; while `n` could stand to be a bit bigger, like `node`. 😁

Reply:
I don't mind `n`, given that its scope is only about a dozen-and-a-half lines (although, it doesn't stand very well in opposition to the other loop variable, `nodeID` 🤷), but, I don't much like `hasNonObviatedDependencies`, not only for its length but also because I think it's defined as "true for negative" which makes it somewhat harder to grok.  [I also hit on `isReady`](https://github.com/arcalot/go-dgraph/pull/15#discussion_r1638757489)....  🙂

Reply:
This section of code was simplified as suggested by Webb.